pretix_sumup/payment.py

```diff
@@ -68,10 +147,32 @@ def settings_form_clean(self, cleaned_data: dict):
         cleaned_data = super().settings_form_clean(cleaned_data)
         access_token = cleaned_data.get("payment_sumup_access_token")
         if access_token is not None and access_token != SECRET_REDACTED:
-            merchant_code = sumup_client.validate_access_token_and_get_merchant_code(
-                access_token
+            merchant_name, merchant_code = (
+                sumup_client.validate_access_token_and_get_merchant_code(access_token)
             )
             cleaned_data["payment_sumup_merchant_code"] = merchant_code
+            cleaned_data["payment_sumup_merchant_name"] = merchant_name
+
+        # Validate Google Pay settings
+        apms_enabled = cleaned_data.get("payment_sumup_enable_apms", False)
+        google_pay_enabled = (
+            cleaned_data.get("payment_sumup_google_pay_enabled", False) and apms_enabled
+        )
+        if google_pay_enabled:
+            merchant_id = cleaned_data.get("payment_sumup_google_pay_merchant_id")
+            merchant_name = cleaned_data.get("payment_sumup_merchant_name")
+
+            if not merchant_id:
+                raise forms.ValidationError(
+                    _("Google Pay Merchant ID is required when Google Pay is enabled.")
+                )
+            if not merchant_name:
```

Comment:
Shouldn't the `merchant_name` always be available if a valid sumup token was provided

Reply:
Only edge case where this won't hold true would be if updating the plugin with an existing configuration?